### ISSUE TITLE
Would Hevm accept a PR that adds evm-opcodes as dependency?

### ISSUE THREAD

**Issue by sshine** [dapphub/dapptools#486]
Hello there,

tl;dr: If I fork Hevm and change the [EVM.Op](https://github.com/dapphub/dapptools/blob/master/src/hevm/src/EVM/Op.hs) module to conform to use [evm-opcodes](https://github.com/sshine/hs-evm-opcodes/), would you be interested in merging this?

A while ago I made the Haskell package [evm-opcodes](https://github.com/sshine/hs-evm-opcodes/) intended for interop with a compiler that targets EVM directly. I didn't get around to replace the compiler's original opcode data type, since the reworking of the code generator spawned more ideas but fewer directions. Those ideas are still taking shape, and a lot of those suggest that a native EVM back-end is preserved (goals related to efficiency, correctness).

As stated, one of the two purposes of the package is to improve interoperability with other modules, of which Hevm is the most obvious, since it allows for simulations during unit and property testing of high-level contracts.

Note: evm-opcodes is not on Hackage yet. I would prefer to increase the test coverage of its only algorithm before doing so. Integrating with Hevm would give me the motivation to finalize this. If you have opinions on the design of this library, feel free to air them as it currently, like hevm, does not have any dependencies on Hackage.

Note: The Network.Ethereum... namespace was chosen from the [web3](https://hackage.haskell.org/package/web3) Haskell package which was the only Ethereum-related project at the time of making this library. It seems that this package has not received any love in a while. Since then I believe less deep namespaces have become more popular, so EVM.Opcode might be a more fitting name.

**Comment by MrChico** [comment]
Hmm, I appreciate the interest but I don't think I see the purpose of substituting the EVM.Op module for your library.... Its very convenient for us to be able have the opcodes under our own repo so its easy for us to change and keep up to date, and its not much maintainance burden for this tiny module

**Comment by sshine** [comment]
A language based on [Bahr et al. (2015)](https://bahr.io/pubs/entries/bahr15icfp.html). There are two implementations, [1](https://github.com/HIPERFIT/contracts) (Coq), [2](github.com/sshine/lira) (EVM backend), but I'd like to make true statements about the EVM code being generated, and not just the formal semantics. Some of those could be backed by Hevm-powered property tests, but it's all experimental to me.
